api/v1alpha1/controlplane_types.go

```diff
@@ -90,12 +99,25 @@ type SecretReference struct {
 	// `name` is the name of the secret.
 	// Required
 	Name string `json:"name"`
-	// Required
+	// +optional
 	Key string `json:"key"`
 	// Required
 	InClusterKey string `json:"inClusterKey"`
 }
 
+// We do not use ObjectReference as its use is discouraged in favor of a locally defined type.
+// See ObjectReference in https://github.com/kubernetes/api/blob/master/core/v1/types.go
```

Comment:
I think that this comment should explain what data the referenced Secret is expected to hold.

Reply:
done

Reply:
Not pushed to GitHub yet, as far as I can tell.

Reply:
working on adding a e2e test for this feature, as suggested, will push it later today